lunes_cms/cms/models/feedback.py

```diff
@@ -0,0 +1,34 @@
+# from django.contrib.contenttypes.fields import GenericForeignKey
+from django.contrib.contenttypes.models import ContentType
+from django.utils.translation import ugettext_lazy as _
+from django.db import models
+
+
+class Feedback(models.Model):
+    """
+    Contains a title, a description and a reference to the matter of feedback
+    """
+
+    title = models.CharField(max_length=255)
+    description = models.TextField()
+    discipline = models.ForeignKey(ContentType, on_delete=models.CASCADE)
+    document = models.ForeignKey(ContentType, on_delete=models.CASCADE)
+    training_set = models.ForeignKey(ContentType, on_delete=models.CASCADE)
+    # options = models.GenericForeignKey("general", discipline, document, training_set)
+    # matter = models.CharField(max_length=12, choices=options, default=options[0])
```

Comment:
```suggestion
    content_type = models.ForeignKey(
        ContentType,
        limit_choices_to=models.Q(
            app_label="cms", model__in=["discipline", "trainingset", "document"]
        ),
        on_delete=models.CASCADE,
        verbose_name=_("content type"),
        help_text=_("The content type this feedback entry refers to."),
    )
    object_id = models.PositiveIntegerField(
        verbose_name=_("object id"),
        help_text=_("The id of the object this feedback entry refers to."),
    )
    content_object = GenericForeignKey("content_type", "object_id")
    content_object.short_description = _("refers to")

    comment = models.TextField(
        verbose_name=_("comment"),
    )

    read_by = models.ForeignKey(
        settings.AUTH_USER_MODEL,
        null=True,
        blank=True,
        on_delete=models.SET_NULL,
        related_name="feedback",
        verbose_name=_("marked as read by"),
        help_text=_(
            "The user who marked this feedback as read. If the feedback is unread, this field is empty."
        ),
    )

    created_date = models.DateTimeField(
        auto_now_add=True,
        verbose_name=_("submitted on"),
        help_text=_("The time and date when the feedback was submitted."),
    )
```
I think you misunderstood how the `GenericForeignKey` works, you don't need one field per model you want to link, but you need one field to store the type of the object (`content_type`) and one field to store the id of the object (`object_id`). Those two fields combined yields the generic foreign key `content_object` which can then be used like a regular foreign key (you can set e.g. `feedback.content_object = Discipline.objects.first()` and Django will automatically set `feedback.content_type` to `discipline` and `feedback.object_id` to the id of the first discipline). So the beauty of this feature is that independently of how many different content types you want to reference, you only need the two database fields and the one generic foreign key which handles the conversion.

Apart from that, I copied + pasted a few other fields from our Integreat feedback model: `read_by` and `created_date`.